lib/libutils/ext/include/util.h

```diff
@@ -89,8 +89,9 @@
 /* x has to be of an unsigned type */
 #define IS_POWER_OF_TWO(x) (((x) != 0) && (((x) & (~(x) + 1)) == (x)))
 
-#define ALIGNMENT_IS_OK(p, type) \
-	(((uintptr_t)(p) & (__alignof__(type) - 1)) == 0)
+#define IS_ALIGNED(x, a)		(((x) & ((a) - 1)) == 0)
+#define IS_ALIGNED_WITH_TYPE(x, type) \
```

Comment:
Now we have two alignment checking macros. I'm wondering if we can make this macro more safe by adding a check that the `type` argument really is a type and not `sizeof(type)` or some other easy mistake.
How about something like:
``` C
#define IS_ALIGNED_WITH_TYPE(x, type) \
        (__extension__({ \
                type __is_aligned_y; \
                IS_ALIGNED((uintptr_t)(x), __alignof__(__is_aligned_y)); \
        }))
```